client/packages/invoices/src/Returns/api/hooks/document/useInsertInboundReturn.ts

```diff
@@ -0,0 +1,20 @@
+import { useQueryClient, useMutation } from '@openmsupply-client/common';
+import { useReturnsApi } from '../utils/useReturnsApi';
+
+export const useInsertInboundReturn = () => {
+  const queryClient = useQueryClient();
+  // const navigate = useNavigate();
+  const api = useReturnsApi();
+  return useMutation(api.insertInboundReturn, {
```

Comment:
Do you think there is a way use the return from mutation as the new cache for the return ? Or is it cleaner to just do another query by invalidating cache ?

This comment is not just for this mutation, but other mutations that return new 'state', and actually the return from API is kind of made to avoid extra API calls

Reply:
I've wondered about this too. We could use [setQueryData](https://tanstack.com/query/v4/docs/framework/react/guides/prefetching#manually-priming-a-query) to manually insert the mutation result into the cache. that would save a round trip to the api 🤔

Reply:
Oo this would be rad!